test/parallel/test-util-inspect-proxy.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const util = require('util');
+
+const target = {};
+const handler = {
+  get: function() { throw new Error('Getter should not be called'); }
+};
+const proxyObj = new Proxy(target, handler);
+
+assert(util.isProxy(proxyObj));
+// Inspecting the proxy should not actually walk it's properties
+assert.doesNotThrow(() => util.inspect(proxyObj));
+
+const details = process.binding('util').getProxyDetails(proxyObj);
```

Comment:
Similarly, it would be nice to verify the return value when this is called without a proxy object.